rmgpy/data/thermo.py

```diff
@@ -214,16 +217,19 @@ def removeThermoData(thermoData1, thermoData2, groupAdditivity=False):
     if len(thermoData1.Tdata.value_si) != len(thermoData2.Tdata.value_si) or any([T1 != T2 for T1, T2 in zip(thermoData1.Tdata.value_si, thermoData2.Tdata.value_si)]):
         raise Exception('Cannot take the difference between these ThermoData objects due to their having different temperature points.')
 
+    testZero = abs(thermoData2.H298.value_si)+abs(thermoData2.S298.value_si) # Used to check if all of the entries in thermoData1 are zero
     for i in range(thermoData1.Tdata.value_si.shape[0]):
         thermoData1.Cpdata.value_si[i] -= thermoData2.Cpdata.value_si[i]
+        testZero += abs(thermoData2.Cpdata.value_si[i]) # Used to check if all of the entries in thermoData1 are zero
     thermoData1.H298.value_si -= thermoData2.H298.value_si
     thermoData1.S298.value_si -= thermoData2.S298.value_si
 
     if groupAdditivity:
-        if thermoData1.comment:
-            thermoData1.comment += ' - {0}'.format(thermoData2.comment)
-        else:
-            thermoData1.comment = 'Thermo group additivity estimation: ' + ' - {0}'.format(thermoData2.comment)
+        if testZero !=0: # Used to check if all of the entries in thermoData1 are zero
+            if thermoData1.comment:
+                thermoData1.comment += ' - {0}'.format(thermoData2.comment)
```

Comment:
This gives you an unsimplified comment at the end. A better way to do it is to regular expression substitute it out. We probably need to think about the regex a bit more, but off-the-cuff suggestion:

`thermoData1.comment = re.sub(re.escape(thermoData2.comment)+'\b', '', thermoData1.comment)`